src/incremental/compareCIL.ml

```diff
@@ -69,7 +69,13 @@ let compareCilFiles ?(eq=eq_glob) (oldAST: file) (newAST: file) =
 
   let addGlobal map global  =
     try
-      GlobalMap.add (identifier_of_global global) global map
+      let gid = identifier_of_global global in
+      let gid_to_string gid = match gid.global_t with
+        | Var -> "Var " ^ gid.name
+        | Decl -> "Decl " ^ gid.name
+        | Fun -> "Fun " ^ gid.name
+        | _ -> raise (NoGlobalIdentifier global) in
+      if GlobalMap.mem gid map then failwith ("Duplicate global identifier: " ^ gid_to_string gid) else GlobalMap.add gid global map
```

Comment:
Actually with OCaml hashtables one should almost always use `replace` instead of `add` since the shadowing behavior is surprising, memory leaky and inefficient (for even representing multimaps, see #614). Then it wouldn't possible to store two globals for the same name.
Of course that wouldn't solve anything here, it would just override the old one.

Reply:
The `GlobalMap` is created with OCamls `Map.Make` functor and therefore does not have a replace function as a `Hashtbl` would have. The documentation indicates that `add` has an overwriting behavior if a key already existed. Should we switch to a `Hashtbl`?

Reply:
Ahh, nevermind then. I thought it's a hashtable like other mutated data structures around these parts. As long as it's not a bottleneck, it should be fine.